### PR TITLE
Core: Fix the incorrectly log description in BinPackStrategy

### PR DESCRIPTION
**Pull request by Zhangg7723**
This PR fixes a log error for rewrite parameter max-file-size-bytes, when I set the target-file-size-bytes less than max-file-size-bytes, spark throwed a checkAgument exception "Cannot set max-file-size-bytes is greater than or equal to target-file-size-bytes, all files written will be larger than the threshold", it's puzzling, so I fix this error.

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/actions/BinPackStrategy.java [hunk 1]
```diff
@@ -259,7 +259,7 @@ private void validateOptions() {
         MIN_FILE_SIZE_BYTES, RewriteDataFiles.TARGET_FILE_SIZE_BYTES, minFileSize, targetFileSize);
 
     Preconditions.checkArgument(targetFileSize < maxFileSize,
-        "Cannot set %s is greater than or equal to %s, all files written will be larger than the threshold, %d >= %d",
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/actions/BinPackStrategy.java:262]: I think the correct fix is to reverse `MAX_FILE_SIZE_BYTES` and `RewriteDataFiles.TARGET_FILE_SIZE_BYTES`. Leaving "or equal to" doesn't work.

Comment (Zhangg7723) [reply core/src/main/java/org/apache/iceberg/actions/BinPackStrategy.java:262]: If  MAX_FILE_SIZE_BYTES equals to TARGET_FILE_SIZE_BYTES, argument check still works and throws this exception. Anyway, there is a mistake I forget to fix "%d >= %d" with parameters, I will reverse targetFileSize and maxFileSize later.
thanks for your comment

Comment (rdblue) [reply core/src/main/java/org/apache/iceberg/actions/BinPackStrategy.java:262]: I think this is incorrect for the equals case. Can you revert the message changes and swap the argument order?

Comment (Zhangg7723) [reply core/src/main/java/org/apache/iceberg/actions/BinPackStrategy.java:262]: OK, I will follow your advice, swap the argument order.

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @Zhangg7723! Looks good. I'll merge when tests are passing.

**Pull request merged**